### PR TITLE
authorization

### PR DESCRIPTION
**Pull request by quanglx-noovado**
## Related Tickets
- [#38912](https://edu-redmine.sun-asterisk.vn/issues/38912)

## WHAT (optional)
Authorization

### CODE REVIEW

**Review by hoanghuee — APPROVED** [review]

**Review by hoanghuee — APPROVED** [review]

**Comment by quanglx-noovado** [comment]
ready

**Review by huyenttm-0901 — COMMENTED** [review]
conflict

**Comment by quanglx-noovado** [comment]
ready

File: app/Policies/DocumentPolicy.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class DocumentPolicy
+{
+    use HandlesAuthorization;
+
+    public function delete(User $user, Document $document)
+    {
+        $author = $document->uploadBy;
+        return $user->id === $author->id;
```
Comment (huyenttm-0901) [inline app/Policies/DocumentPolicy.php]: convention

File: routes/web.php [hunk 2]
```diff
@@ -26,14 +26,18 @@
...
     Route::post('unmark/{id}', 'DocumentController@unmark')->name('documents.unmark');
-    Route::post('download/{id}', 'DocumentController@download')->name('documents.download');
-    Route::get('/user/buycoin', 'UserController@buyCoin')->name('buy-coin');
-    Route::get('/payment', 'UserController@payment')->name('payment');
+    Route::post('download/{id}', 'DocumentController@download')
+        ->name('documents.download')->middleware('auth');
+    Route::get('/user/buycoin', 'UserController@buyCoin')
+        ->name('buy-coin')->middleware('auth');
+    Route::get('/payment', 'UserController@payment')
+        ->name('payment')->middleware('auth');
```
Comment (huyenttm-0901) [inline routes/web.php]: payment có thông tin thanh toán của user, dùng get à em?

**Comment by quanglx-noovado** [comment]
ready

**Review by huyenttm-0901 — APPROVED** [review]

**Pull request merged**